src/infrastructure/plugins/shared_data.ts

```diff
@@ -0,0 +1,23 @@
+import { ISharedData } from '../../interfaces/sharedData.interface';
+import { Server } from '@hapi/hapi';
+
+module.exports = {
+  name: 'SharedData',
```

Comment:
i personally think this plugin is not going workout when you the scale the application horizontally. i prefer not to use any cache plugin as we are using elastic search. if you still want to have one, please think about using redis or memcache